codes/classical/q-ary_digits/rank_modulation.yml

```diff
@@ -0,0 +1,43 @@
+#######################################################
+## This is a code entry in the error correction zoo. ##
+##       https://github.com/errorcorrectionzoo       ##
+#######################################################
+
+
+# code id, physical, logical are all lower case
+code_id: rank_modulation
+physical: groups
+logical: q-ary_digits
+
+
+name: 'Rank modulation code'
+introduced: '\cite{doi:10.1109/ISIT.2008.4595285}'
+
+
+description: |
+    Rank modulation codes are a family of codes which encode an alphabet of size \(q\) into a set \(S_n\) of permutations of \([n]=(1,2,...,n)\).
```

Comment:
First sent: Also known as a \textit{code in permutations}.

Reply:
Note that there is no "alphabet of size q". The codes encode a set of M messages (a finite set of size M). There is no q that goes with them. To every message the encoder associates a permutation on n elements, where n can be thought of as the length of the code.